tinyiiod.c

```diff
@@ -62,160 +74,219 @@ int tinyiiod_read_command(struct tinyiiod *iiod)
 	return ret;
 }
 
+/***************************************************************************//**
+ * @brief tinyiiod_read_char
+*******************************************************************************/
 char tinyiiod_read_char(struct tinyiiod *iiod)
 {
 	char c;
 
-	iiod->ops->read(&c, 1);
+	iiod->ops->read(&iiod->instance_id, &c, 1);
+
 	return c;
 }
 
-void tinyiiod_read(struct tinyiiod *iiod, char *buf, size_t len)
+/***************************************************************************//**
+ * @brief tinyiiod_read
+*******************************************************************************/
+int32_t tinyiiod_read(struct tinyiiod *iiod, char *buf, size_t len)
 {
-	iiod->ops->read(buf, len);
+	return iiod->ops->read(&iiod->instance_id, buf, len);
 }
 
-ssize_t tinyiiod_read_line(struct tinyiiod *iiod, char *buf, size_t len)
+/***************************************************************************//**
+ * @brief tinyiiod_read_line
+*******************************************************************************/
+int32_t tinyiiod_read_line(struct tinyiiod *iiod, char *buf)
```

Comment:
Do we need this wrapper anymore? We could just call iiod->ops->read_line(&iiod->instance_id, buf); and just use the iiod object.